firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexManager.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import com.google.firebase.firestore.model.ResourcePath;
+import java.util.List;
+
+/**
+ * Represents a set of indexes that are used to execute queries efficiently.
+ *
+ * <p>Currently the only index is a [collection id] => [parent path] index, used to execute
+ * Collection Group queries.
+ */
+public interface IndexManager {
+  /**
+   * Creates an index entry mapping the collectionId (last segment of the path) to the parent path
+   * (either the containing document location or the empty path for root-level collections). Index
+   * entries can be retrieved via getCollectionParents().
+   */
+  void addToCollectionParentIndex(ResourcePath collectionPath);
```

Comment:
Question: do we never remove from the index?

Reply:
Correct. I'll add that to the comments!